### PR TITLE
Refactor FastApi Dag and DagRun endpoints tests

### PR DESCRIPTION
**Pull request by rawwar**
Related to #42866

### CODE REVIEW

**Comment by rawwar** [comment]
@pierrejeambrun, please review

File: dev/tests_common/pytest_plugin.py [hunk 1]
```diff
@@ -118,6 +118,7 @@
 ).resolve()
 AIRFLOW_TESTS_DIR = AIRFLOW_SOURCES_ROOT_DIR / "tests"
 
+os.environ["AIRFLOW_TESTS_DIR"] = os.fspath(AIRFLOW_TESTS_DIR)
```
Comment (pierrejeambrun) [inline dev/tests_common/pytest_plugin.py]: It doesn not seem that this env variable is used here, is it ?

**Review by pierrejeambrun — APPROVED** [review]
Just a question before we merge, otherwise looking good.

**Comment by pierrejeambrun** [comment]
Test failure is unrelated,  we need to wait and rebase on top of https://github.com/apache/airflow/pull/43002

File: dev/tests_common/pytest_plugin.py (hunk 1 above)
Comment (rawwar) [reply dev/tests_common/pytest_plugin.py]: I should have removed it. Removing it now

**Comment by rawwar** [comment]
> Test failure is unrelated, we need to wait and rebase on top of #43002

Done.

**Review by ephraimbuddy — APPROVED** [review]

**Pull request merged**